scm/content.go

```diff
@@ -16,10 +19,20 @@ type (
 	// ContentParams provide parameters for creating and
 	// updating repository content.
 	ContentParams struct {
-		Ref     string
-		Branch  string
-		Message string
-		Data    []byte
+		Branch    string        `json:"branch,omitempty" description:"The branch name. Default: the repository’s default branch."`
+		Message   string        `json:"message,omitempty" description:"The commit message."`
+		Data      []byte        `json:"data,omitempty" description:"The new file content, using Base64 encoding."`
+		SHA       string        `json:"sha,omitempty" description:"Required if you are updating a file. The blob SHA of the file being replaced."`
+		Author    *CommitAuthor `json:"author,omitempty" description:"The author of the file."`
```

Comment:
to ensure consistency with the rest of the codebase, these should not be pointers

Reply:
replace with https://github.com/drone/go-scm/blob/master/scm/git.go#L41